src/scores/categorical/contingency_impl.py

```diff
@@ -271,14 +271,50 @@ def sensitivity(self):
         """
         return self.probability_of_detection()
 
-    def specificity(self):
+    def specificity(self) -> XarrayLike:
```

Comment:
Have we tested that this works with xr.Datasets or just Datarrays?
If the former, we need to update unit tests, if the latter, this needs to change to `-> xr.Dataarray`

Reply:
Following the discussion on another MR, we should move to xr.DataArray here.

Reply:
Thanks for the review. I have changed both specificity and true negative rate to xr.DataArray, and also added return type for both.